Softeq.XToolkit.Common/Helpers/Hashing/IHashCalculator.cs

```diff
@@ -0,0 +1,25 @@
+// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+namespace Softeq.XToolkit.Common.Helpers.Hashing
+{
+    /// <summary>
+    /// Common interface for Hash function implementations
+    /// </summary>
+    public interface IHashCalculator
+    {
+        /// <summary>
+        ///     Adds value to use in hash code calculations.
+        /// </summary>
+        /// <returns>IHashCalculator reference to use in chained calls</returns>
+        /// <typeparam name="T">Generic type of the value</typeparam>
+        /// <param name="value">Value to use in hash code calculations</param>
+        IHashCalculator Using<T>(T value);
```

Comment:
`Using` can be named to `Add`

Reply:
The idea behind naming was to make it more fluent, similar to speech. For instance, "Calculate hashcode using x,y,z" transforms into:
```
Hash.Get().Using(x).Using(y).Using(z).Calculate();
```
Another alternative was to use "of" instead of "using": "Calculate hashcode of x,y,z"..

The only thing that does not really fit it is "Get()" method. But again, the idea was that using hash-function abstraction (`interface IHashCalculator`) allows us to prepare different implementations of hash-functions for user to choose: FNV, Shift-And-Xor, Bernstein and so on:
```
Hash.FNV().Using(x).Using(y).Using(z).Calculate();
Hash.ShiftAndXor().Using(x).Using(y).Using(z).Calculate();
```
The problem was, I didn't find the name of algorithm that we are using right now, so I decided to give it default name of `Get()` for now. But I guess it could be renamed to `Default()`. The only thing I'm not sure of is to make it property. It is basically a factory method, that creates new instance of IHashCodeCalculator on each call.